monitoring/ping.go

```diff
@@ -0,0 +1,263 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"github.com/gravitational/ttlmap"
+	"time"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/gravitational/trace"
+	serf "github.com/hashicorp/serf/client"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// pingCheckerID specifies the check name
+	pingCheckerID = "ping-checker"
+	// slidingWindowSize specifies the number of retained check results
+	slidingWindowSize = 10
+	// statsTTLPeriod specifies how long check results will be kept before being dropped
+	statsTTLPeriod = 1 * time.Hour
+	// pingRoundtripMinimum set the minim value that can be recorded
+	pingRoundtripMinimum = 0 * time.Second
+	// pingRoundtripMaximum set the maximum value that can be recorded
+	pingRoundtripMaximum = 10 * time.Second
+	// pingRoundtripSignificativeFigures specifies how many decimals should be recorded
+	pingRoundtripSignificativeFigures = 3
+	// pingRoundtripThreshold sets the RTT threshold
+	pingRoundtripThreshold = 15 * time.Millisecond
+	// pingRoundtripQuantile sets the quantile used while checking Histograms against Rtt results
+	pingRoundtripQuantile = 95.0
+)
+
+// NewPingChecker implements and return an health.Checker
+func NewPingChecker(serfRPCAddr string, serfMemberName string) health.Checker {
+	roundtripLatencyTTLMap, err := ttlmap.New(int(statsTTLPeriod.Seconds()))
```

Comment:
I got it wrong and thought it was the maximum TTL possible.

I'm not sure what to set here though cause this would be the number of "other" nodes in the cluster but this number can vary over time and there's no upper limit that I know of in Gravity.

Maybe I could connect to Serf and fetch the current number of nodes and use that... but then again it may change over time :man_shrugging:

I guess I can use the number from Serf initially and then update it later before Set() calls?